hail/src/main/scala/is/hail/expr/ir/SpecializedArrayBuilders.scala

```diff
@@ -406,4 +406,46 @@ class BooleanArrayBuilder(initialCapacity: Int) extends MissingArrayBuilder(init
       i += 1
     }
   }
+}
+
+class ByteArrayArrayBuilder(initialCapacity: Int) {
```

Comment:
Can't you use `ArrayBuilder[Array[Byte]]`?  The other specializations are for primitive types to avoid boxing.

Reply:
I tried, but there's something weird about parameterized types (maybe even for unspecialized versions of specialized parameters) that was causing an error to be thrown.